lib/api/redfish-1.0/event-service.js

```diff
@@ -0,0 +1,266 @@
+                tool.settings.port = parse.port;
+                record.MemberId = event.node;
+                record.EventType = "StatusChange";
+                record.EventId = event.reading.sensorId;
+                record.EventTimestamp = new Date().toISOString();
+                record.Message = event.reading.entryIdName + ', statesAsserted: ' + 
+                    event.reading.statesAsserted.toString();
+                record.MessageId = "StatusChange.1.0." + 
+                    event.reading.sensorId.replace(/[^A-Za-z0-9.]/g,'');
+                return schemaApi.validate(record, 'Event.1.0.0.json#/definitions/EventRecord')
+                .then(function(result) {
+                    if(result.error) {
+                        throw new Error(result.error);
+                    }
+                    return sendClient(tool, retryAttempts, record);
+                });
+            });
+        }
+    });
+};
+
+var eventServiceRoot = controller(function(req, res) {
+    var options = redfish.makeOptions(req, res);
+    return Promise.resolve()
+    .then(function() {
+        options.timeoutInSeconds = timeoutInSeconds;
+        options.retryAttempts = retryAttempts;
+        return redfish.render('redfish.1.0.0.eventservice.1.0.0.json',
+                'EventService.1.0.0.json#/definitions/EventService',
+                options);
+    })
+    .catch(function(error) {
+        return redfish.handleError(error, res);
+    });
+});
+
+var getEventsCollection = controller(function(req, res) {
+    var options = redfish.makeOptions(req, res);
+    return Promise.resolve()
+    .then(function() {
+        options.subscriptions = _.values(eventSubscriptions);
+        return redfish.render('redfish.1.0.0.eventdestinationcollection.json',
+                'EventDestinationCollection.json#/definitions/EventDestinationCollection',
+                options);
+    })
+    .catch(function(error) {
+        return redfish.handleError(error, res);
+    });
+});
+
+var createSubscription = controller(function(req, res) {
+    var options = redfish.makeOptions(req, res);
+    var event = req.swagger.params.payload.value;
+    return schemaApi.validate(event, 
+        'EventDestination.1.0.0.json#/definitions/EventDestination'
+    )
+    .then(function validatePayload(result) {
+        if(result.error) {
+            throw new Error(result.error);
+        }
+        var index = _.findIndex(_.values(eventSubscriptions), function(subscription) {
+            return subscription.Destination === event.Destination;
+        });
+        
+        if(index > -1) {
+            var err = new Errors.BaseError('EventDestination Exists');
+            err.status = 409;
+            throw err;
+        }
+        event.Id = uuid('v4');
+        _.set(eventSubscriptions, event.Id, event);
+        return event;
+    })
+    .then(function(output) {
+        eventListener(eventCallback);
```

Comment:
@benbp I tested the typical error scenarios from the eventListener such as sendClient() and schemaApi.validate() exceptions.  Most of the code there is synchronous and not much to handle in terms of errors.   Outside of failing to send the event record to the client or error validation what should we be handling?